resources/js/Pages/Results.vue

```diff
@@ -19,26 +21,47 @@
     import { PropType } from 'vue';
 
     import { Head } from '@inertiajs/inertia-vue';
+    import { Link as WikitLink } from '@wmde/wikit-vue-components';
 
     import MismatchesTable from '../Components/MismatchesTable.vue';
-    import Mismatch from '../types/Mismatch';
+    import Mismatch, {LabelledMismatch} from '../types/Mismatch';
     import defineComponent from '../types/defineComponent';
 
     interface Result {
         [qid: string]: Mismatch[]
     }
 
+    interface LabelMap {
+        [entityId: string]: string
+    }
+
     export default defineComponent({
         components: {
             Head,
-            MismatchesTable
+            MismatchesTable,
+            WikitLink
         },
         props: {
             item_ids: Array as PropType<string[]>,
-            results: Object as PropType<Result>
+            results: Object as PropType<Result>,
+            labels: Object as PropType<LabelMap>
+        },
+        methods: {
+            mapLabels(mismatches: Mismatch[]): LabelledMismatch[]{
+                return mismatches.map(mismatch => ({
+                    property_label: this.labels[mismatch.property_id],
+                    value_label: this.labels[mismatch.wikidata_value] || null,
+                    ...mismatch
+                }));
```

Comment:
ok, this looks equally scary as the `reduce` function below, but basically adds a `property_label` and `value_label` field (or `null`, if not applicable?) to each mismatch, while leaving the rest of the mismatch fields untouched, right? Nice. Maybe also worth a comment? Or do you think the code speaks for itself here?

Also, the method name: wouldn't `addLabels` actually be a better fit here, because `map` is the really technical activity of iterating through the mismatches and applying some function to each element, rather than the conventional "map A to B"?

Reply:
> Maybe also worth a comment? Or do you think the code speaks for itself here?

To me the code speaks for itself, this is a pretty common pattern in JS. But, I'm not a big fan of implicit knowledge, so if you feel like a comment is needed, that is reason enough for me to add one :smiley: 

> Also, the method name: wouldn't addLabels actually be a better fit here

Sure, yeah, why not all for clarity.

Reply:
Done